auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```diff
@@ -318,82 +323,83 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
-                    "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
-                            "Any previously stored content is now lost. Please, try saving the credentials again.",
-                    e
+                decryptCallback = null
+                return@execute
+            } catch (e: CryptoException) {
+                //If keys were invalidated, existing credentials will not be recoverable.
+                clearCredentials()
+                callback.onFailure(
+                    CredentialsManagerException(
+                        "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
+                                "Any previously stored content is now lost. Please, try saving the credentials again.",
+                        e
+                    )
                 )
+                decryptCallback = null
+                return@execute
+            }
+            val bridgeCredentials = gson.fromJson(json, OptionalCredentials::class.java)
+            /* OPTIONAL CREDENTIALS
+             * This bridge is required to prevent users from being logged out when
+             * migrating from Credentials with optional Access Token and ID token
+             */
+            val credentials = Credentials(
+                bridgeCredentials.idToken.orEmpty(),
+                bridgeCredentials.accessToken.orEmpty(),
+                bridgeCredentials.type.orEmpty(),
+                bridgeCredentials.refreshToken,
+                bridgeCredentials.expiresAt ?: Date(),
+                bridgeCredentials.scope
+            )
+            val cacheExpiresAt = storage.retrieveLong(KEY_CACHE_EXPIRES_AT)
+            val expiresAt = credentials.expiresAt.time
+            val hasEmptyCredentials =
+                TextUtils.isEmpty(credentials.accessToken) && TextUtils.isEmpty(credentials.idToken) || cacheExpiresAt == null
+            if (hasEmptyCredentials) {
+                callback.onFailure(CredentialsManagerException("No Credentials were previously set."))
+                decryptCallback = null
+                return@execute
+            }
+            val hasEitherExpired = hasExpired(cacheExpiresAt!!)
+            val willAccessTokenExpire = willExpire(expiresAt, minTtl.toLong())
+            val scopeChanged = hasScopeChanged(credentials.scope, scope)
+            if (!hasEitherExpired && !willAccessTokenExpire && !scopeChanged) {
+                callback.onSuccess(credentials)
+                decryptCallback = null
+                return@execute
+            }
+            if (credentials.refreshToken == null) {
+                callback.onFailure(CredentialsManagerException("No Credentials were previously set."))
+                decryptCallback = null
+                return@execute
+            }
+            Log.d(TAG, "Credentials have expired. Renewing them now...")
```

Comment:
Is it ok to leave this `Log` in?

Reply:
Since this is a 'debug' level log I guess this should be helpful for the developers integrating our SDK. Shouldn't affect the prod.